File: common/jwt-core/src/main/java/com/halcyon/tinder/jwtcore/JwtAuthentication.java
package com.halcyon.tinder.jwtcore;

import java.util.Collection;
import java.util.Collections;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JwtAuthentication implements Authentication {

    private boolean authenticated;
    private String phoneNumber;
    private String token;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return Collections.emptySet();
    }

    @Override
    public Object getCredentials() {
        return null;
    }

    @Override
    public Object getDetails() {
        return phoneNumber;
    }

    @Override
    public Object getPrincipal() {
        return phoneNumber;
    }

    @Override
    public String getName() {
        return phoneNumber;
    }
}

File: common/jwt-core/src/main/kotlin/com/halcyon/tinder/jwtcore/JwtAuthentication.kt
package com.halcyon.tinder.jwtcore

import org.springframework.security.core.Authentication
import org.springframework.security.core.GrantedAuthority

data class JwtAuthentication(
    private var authenticated: Boolean = true,
    var phoneNumber: String? = null,
    var token: String? = null,
) : Authentication {

    override fun getName(): String ?= phoneNumber
    override fun getAuthorities(): Collection<GrantedAuthority> = emptySet()
    override fun getCredentials(): Any ?= null
    override fun getDetails(): Any ?= null
    override fun getPrincipal(): Any ?= phoneNumber
    override fun isAuthenticated(): Boolean = authenticated
    override fun setAuthenticated(isAuthenticated: Boolean) {
        authenticated = isAuthenticated
    }

}